Port improvements from ParallelProcessingTools

oschulz opened the pull request.

oschulz committed:
Port improvements from ParallelProcessingTools

oschulz committed:
Increase package version to v2.0.0

Removing ElasticManager.printing_kwargs may be considered breaking,
it only affects show(::ElasticManager) but it was documented in
the README.

oschulz committed:
Document elastic_worker

Some use cases need to call elastic_worker, so it should be part of the
API.

Also rename stdout_to_master to forward_stdout, it's clearer.
Non-breaking since stdout_to_master was never documented.

oschulz committed:
Replace ClusterManagers by ElasticClusterManager in README and license file

oschulz committed:
Document ElasticManager

oschulz committed:
Remove ElasticManager field printing_kwargs

The name of the field is confusing. It's much cleaner for the users to call
`get_connect_cmd` directly. Workers may be added in different ways to the same
ElasticManager and require a different startup. Also, `get_connect_cmd` was not
using printing_kwargs for it's kwargs by default, which was inconsistent.

DilumAluthge commented:
Tbh I'm not terribly familiar with the `ElasticManager` code. @Moelf Would you be able to review?

oschulz commented:
Test are fixed now.

oschulz commented:
Still tracking down an issue with persistent tasks.

oschulz committed:
Don't export export launch, manage, kill, init_worker, connect

These are owned by Distributed, there's no reason to export them
from ElasticClusterManager.

oschulz commented:
> Still tracking down an issue with persistent tasks.

Still trying to figure out what's going on there ...

oschulz commented:
> Still tracking down an issue with persistent tasks.

It looks like that was just a local artifact of Aqua testing on ParallelProcessingTools.

oschulz commented:
For future reference, `Aqua.test_persistent_tasks(SomePackage)` reports a "persistant tasks" error when the local dev version of `SomePackage` depends on an unreleased version of `SomeOtherPackage` , even if that package is also in "dev-mode" and it has the new version locally - apparently `Aqua.test_persistent_tasks` runs an independent dependency resolution against the registry as part of what it does, and the fact that it then fails to resolve versions results in a "persistent task". Which normally wouldn't have been so confusing, but when testing a package whose tests involve running parallel tasks ...
